packages/atomic/src/components/commerce/atomic-commerce-sort-dropdown/atomic-commerce-sort-dropdown.spec.ts

```diff
@@ -0,0 +1,157 @@
+import {fixture} from '@/vitest-utils/testing-helpers/fixture';
```

Comment:
I really like this: I think this is the right scope for a component test.

One thing I'm wondering is if we should make the `select` method public and test it. And more generally, if we should test these kinds of handlers... We're covering the functionality in the e2e tests, but perhaps it would be good to also cover it in a lower testing layer.

Another open question: in general, should we check that we're calling `bindStateToController` with the expected arg? This would indirectly validate that we're subscribing to / unsubscribing from controller state changes correctly. My reasoning is that if we weren't using the decorator at all, then we'd be subscribing in the `initialize` and unsubscribing in the `disconnectedCallback`, and I believe we would then want to test those behaviors.

Reply:
For the select being public, I think the approach is to test if this.sort.sortBy is called instead of ensure the select method is called. instead we can do something like
```ts
it('should call sort.sortBy when select is changed', async () => {
  await setupElement();
  await locators.select.selectOptions('foo');
  expect(mocks.sort.sortBy).toHaveBeenCalledWith({
    by: 'fields',
    fields: [
      {
        name: 'foo',
      },
    ],
  });
});
```

For the bindStateToController, I don't think it is necessary. The subscription is already tested in `/src/decorators/bind-state.spec.ts`
But i guess we can test the decorator being called with the right params